Add support for SVG images

corny opened the pull request:
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/374adcca-5193-4f3a-85d9-fddf9c3d1777) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/60851399-1fcb-4369-9b4a-bfef61616f09) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/240d52fd-2ffa-40e4-a878-5af905251301) was created.

corny commented:
@laf the current PNG logos are looking aweful on high DPI displays.

I updated my pull request. `foreach (array("svg", "png") as $ext) {` should work with PHP 5.3? I replaced some logos by SVG versions. More work has to done, but PNG is still supported.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/60851399-1fcb-4369-9b4a-bfef61616f09) completed: **2 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/240d52fd-2ffa-40e4-a878-5af905251301) completed: **2 updated code elements**

laf commented:
I think I'd be more comfortable personally if you could add a doc on how to generate svg images to be used otherwise people are going to continue to use png images.

You've got more shorthand array calls which need updating. https://travis-ci.org/librenms/librenms/jobs/187973765

Thanks for contributing btw. You do need to sign the contributors agreement - you removed all of that text above, I've added it back in, please confirm as per the instructions.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9f3f2f35-dadd-4d06-96b4-945e09f157f0) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/10caec9d-4a2b-4129-a993-b466f7b637a5) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9f3f2f35-dadd-4d06-96b4-945e09f157f0) completed: **2 updated code elements**

corny commented:
I added myself to the `CONTRIBUTING.md`, ok thanks for re-adding the text.

There is basically nothing to be aware of when creating SVG images. They are automatically scaled by stylesheets. I grabbed some images from wikipedia and I also cut some. Logos should just not be larger than ~ 20 kb. If the svg is too large, then paths should be simplified (e.g. with Inkscape). Where can I document it?

The SVG version of some logos is even smaller than the PNG.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/10caec9d-4a2b-4129-a993-b466f7b637a5) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/801a7370-d945-44ee-a416-f881c85e7788) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/801a7370-d945-44ee-a416-f881c85e7788) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4990a255-eaa0-4b43-90cb-ab4d8caca88b) was created.

corny commented:
I also changed the second shorthand array and set a height for logo inside of the `<span class="device_icon">`.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4990a255-eaa0-4b43-90cb-ab4d8caca88b) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/73bc05fa-0a1f-434e-82f5-0ef44a803ae9) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/73bc05fa-0a1f-434e-82f5-0ef44a803ae9) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/8e30a7c3-1290-4857-989a-b9a503da2ec5) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/8e30a7c3-1290-4857-989a-b9a503da2ec5) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/49951920-ecdb-4b8e-99bb-766136191e24) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/49951920-ecdb-4b8e-99bb-766136191e24) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/d3989eba-b8d5-462b-a23a-169825906798) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/d3989eba-b8d5-462b-a23a-169825906798) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/bd20dcd0-1cb4-481e-8b44-4861bf005efe) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/bd20dcd0-1cb4-481e-8b44-4861bf005efe) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/6a829263-b8f3-4d97-82c5-891d69c604fd) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/6a829263-b8f3-4d97-82c5-891d69c604fd) completed: **2 updated code elements**

laf commented:
The contributors agreement needs to be signed as per the doc which means commiting a specific message.

As for docs, I'd say add something about the icon here:

http://docs.librenms.org/Developing/Support-New-OS/

and here:

http://docs.librenms.org/Support/FAQ/#how-do-i-add-support-for-a-new-os

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/932db0b9-0045-48c2-8143-f9f758048f14) was created.

corny commented:
Thanks, I added a dedicated commit for the contributors agreement.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/33fcc84e-46ab-497c-a589-2852c10a03c4) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/932db0b9-0045-48c2-8143-f9f758048f14) completed: **2 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/33fcc84e-46ab-497c-a589-2852c10a03c4) completed: **2 updated code elements**

laf commented:
Once you do some docs explaining to users creating new os' that svg is accepted and where to place the files then this will be good to merge imho.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/cf813f43-8e93-4d0b-9ae4-309746434abd) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ee3abda2-133b-45d7-891b-cb6c4fa4ffd8) completed: **2 updated code elements**

corny commented:
I added the docs.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ebdcccd9-2c6d-4f80-8a60-73f2c0341832) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/cf813f43-8e93-4d0b-9ae4-309746434abd) completed: **2 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ebdcccd9-2c6d-4f80-8a60-73f2c0341832) completed: **2 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/47a506aa-05e9-47c0-891a-9fcb6403b45c) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/7ecff64a-cbcd-428c-ab36-e52e487e88b1) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/7ecff64a-cbcd-428c-ab36-e52e487e88b1) completed: **2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/a33360da-a6db-4758-9985-5db21fdd0e82) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/a33360da-a6db-4758-9985-5db21fdd0e82) completed: **2 updated code elements**

corny commented:
I used the [SVG optimizer](https://github.com/svg/svgo) shrink the images by more than 50 % in average.

murrant commented on includes/functions.php:
```diff
@@ -146,9 +146,19 @@ function getImage($device)
+    return 'images/os/' . getImageName($device);
+}
+
+function findImage($basename)
```
Can you rename this to findOsImage() or something?

murrant commented:
I would like to have SVGs for use in larger thatn 32x32 locations for v2, so +1.

The php code could use some refinement.  I think the findImage() function needs to be removed and folded in.  See code notes for more.

murrant commented on includes/functions.php:
```diff
@@ -158,31 +168,34 @@ function getImageName($device, $use_database = true)
-    if (file_exists($config['html_dir'] . '/images/os/' . $device['os'] . '.png')) {
-        return $device['os'];
+    foreach ($possibilities as $icon) {
+        if (!empty($icon)) {
```
You don't need this check if you use is_file() instead of file_exists() (which returns true for directories)

murrant commented on includes/functions.php:
```diff
@@ -158,31 +168,34 @@ function getImageName($device, $use_database = true)
+    foreach ($possibilities as $icon) {
+        if (!empty($icon)) {
+            $name = findImage($icon);
+            if (!empty($name)) {
```
If you bring the findImage code into this function, you won't need the !empty() check here...

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fbed8850-6521-4422-8efc-4ac5dfea7b72) was created.

corny commented:
@murrant thanks for the review. I pushed another commit.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fbed8850-6521-4422-8efc-4ac5dfea7b72) completed: **1 updated code elements**

murrant commented:
Looking good now.

I would prefer the red and blue for the Cisco logo however :)

corny commented:
@murrant The red/blue Cisco logo looks much better. I replaced it.

murrant commented:
@corny This needs a rebase (docs got moved around in another commit)

@librenms/reviewers Are we concerned about supporting IE8 and older?  SVG img tags is IE9+ (and any other browser)

laf commented:
IE 8 is completely end of life. 

I'd be happy to drop support.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/696408d1-0f31-4484-a236-3255f5878549) was created.

corny commented:
I rebased my branch on top of master.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/696408d1-0f31-4484-a236-3255f5878549) completed: **1 updated code elements**

murrant commented:
Ok, I also reviewed the icons.  They all look great, a few deviate in a way that might be concerning.
apple - some might not like the change to the rainbow logo, I'm fine with it.
aruba and mimosa: these could be changed to just the first letter of the logo as they were before.

A good indicator to me is the favicon of a companies website.  Perhaps it would be a good idea to have both icon and logo (but it would be pointless if they are not used).

corny commented:
Yeah, I could not find a nice looking apple logo. Do you prefer another one?
https://commons.wikimedia.org/wiki/File:Apple_logo_black.svg (look at "Other versions")

I can cut the Aruba and Mimosa logos to the first letter. As a small square icon they would be definately easier to be recognized. Other opinions?

murrant commented:
I prefer the rainbow logo over the plain black.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/48c8b6a8-dcb6-470d-8073-014e139f2243) was created.

corny commented:
I rebased my branch on top of master.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/48c8b6a8-dcb6-470d-8073-014e139f2243) completed: **No new issues**

laf commented:
trigger ci

FTBZ commented:
IMO, I don't think that you can use the logo that you prefer. You need to use the correct corporate logo that match the actual design. The Apple rainbow logo isn't used since 1998...

laf commented:
@librenms/reviewers I'm ok with this. Just rebasing it now so it's ready for a merge.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fdc54bca-6cff-4d55-b055-e8871c72597a) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fdc54bca-6cff-4d55-b055-e8871c72597a) completed: **No new issues**

murrant commented:
Did we want to change the Apple icon as FTBZ mentioned?  It would probably be to the monochrome grey one.

laf commented:
I'm not overly fussed but the black one is more modern imho.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/971911ff-23f2-46aa-a810-b49bf045c170) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b48817f7-574e-4cc6-9d0a-b5a348cfa946) was created.

corny commented:
I replaced by Apple logo by the grey one and rebased again.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/971911ff-23f2-46aa-a810-b49bf045c170) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b48817f7-574e-4cc6-9d0a-b5a348cfa946) completed: **No new issues**
